src/hotspot/share/opto/library_call.cpp

```diff
@@ -5968,7 +5968,7 @@ bool LibraryCallKit::inline_cipherBlockChaining_AESCrypt(vmIntrinsics::ID id) {
 
   ciInstanceKlass* instklass_AESCrypt = klass_AESCrypt->as_instance_klass();
   const TypeKlassPtr* aklass = TypeKlassPtr::make(instklass_AESCrypt);
-  const TypeOopPtr* xtype = aklass->as_instance_type();
+  const TypeOopPtr* xtype = aklass->as_instance_type()->cast_to_ptr_type(TypePtr::NotNull);
```

Comment:
I assume you need this because you've changed `as_instance_type` to no longer cast to `TypePtr::NotNull`. Why is that and what about other usages of `as_instance_type`?

Reply:
I looked at other use of as_instance_type and found that there was not need for the cast to not null. As to why, it makes little sense to me that as_instance_type() would return NotNull. It feels error prone.